packages/next/src/server/image-optimizer.ts

```diff
@@ -587,11 +630,30 @@ export async function imageOptimizer(
       'dangerouslyAllowSVG' | 'minimumCacheTTL'
     >
   },
-  isDev: boolean | undefined
-): Promise<{ buffer: Buffer; contentType: string; maxAge: number }> {
+  isDev: boolean | undefined,
+  previousCacheEntry?: IncrementalCacheItem
+): Promise<{
+  buffer: Buffer
+  contentType: string
+  maxAge: number
+  etag: string
+  upstreamEtag: string
+}> {
   const { href, quality, width, mimeType } = paramsResult
-  const upstreamBuffer = imageUpstream.buffer
+  const { buffer: upstreamBuffer, etag: upstreamEtag } = imageUpstream
   const maxAge = getMaxAge(imageUpstream.cacheControl)
+
+  if (shouldUsePreviouslyCachedEntry(imageUpstream, previousCacheEntry)) {
+    // Return the cached optimized image
+    return {
+      buffer: previousCacheEntry.value.buffer,
+      contentType: paramsResult.mimeType,
```

Comment:
Are you sure the `contentType` is correct here?

We have a lot of logic to determine the contentType here:

https://github.com/vercel/next.js/blob/ca735270d95cafccd50410788b59804fc269e293/packages/next/src/server/image-optimizer.ts#L636-L649